www/js/services.js

```diff
@@ -58,7 +58,31 @@ angular.module('emission.services', [])
       })
     };
 })
+.service('ReferHelper', function($http) {
 
+    this.habiticaRegister = function(groupid, successCallback, errorCallback) {
+        window.cordova.plugins.BEMServerComm.getUserPersonalData("/join.group/"+groupid, successCallback, errorCallback);
+    };
+    this.joinGroup = function(groupid, userid) {
+    
+    // TODO:
+    new Promise(function(resolve, reject) {
+        window.cordova.plugins.BEMServerComm.postUserPersonalData("/join.group/"+groupid, "inviter", userid, resolve, reject);
```

Comment:
the way that promises are supposed to be used is that you pass the promise up until you need to use the result somewhere. That allows for maximum flexibility in scheduling.